### PR TITLE
improvements: cast_batch/2 & batch size settings

### PR DESCRIPTION
**Pull request by benoitc**
this PR add the following improvements:

* add `gen_batch_server:cast_batch/2` function: like cast but send a batch of messages instead of one message. This allows the aggregation of multiple batches.
* make `min_batch_size` and `max_batch_size` configurable when starting the  batch.
* add `gen_batch_server:stop/{1, 3}`

### CODE REVIEW

File: test/gen_batch_server_SUITE.erl [hunk 1]
```diff
@@ -145,6 +147,30 @@ cast_many(Config) ->
...
+    Mod = ?config(mod, Config),
+    meck:new(Mod, [non_strict]),
+    meck:expect(Mod, init, fun(Init) -> {ok, Init} end),
+    Args = #{},
+    {ok, Pid} = gen_batch_server:start_link({local, Mod}, Mod, Args, []),
+    Self = self(),
+    meck:expect(Mod, handle_batch,
+                fun(Ops, State) ->
+                        {cast, {put, K, V}} = lists:last(Ops),
+                        ct:pal("cast_batch: batch size ~b~n", [length(Ops)]),
```
Comment (michaelklishin) [inline test/gen_batch_server_SUITE.erl]: Was this left in intentionally?

Comment (benoitc) [reply test/gen_batch_server_SUITE.erl]: @michaelklishin this is similar to the `cast_many` test, but can be removed if needed :-)

Comment (michaelklishin) [reply test/gen_batch_server_SUITE.erl]: I don't know if it was kept there before committing intentionally. We occasionally have such messages in CT suites but only for truly complex tests and with the low priority option. Can we remove them here?

**Comment by benoitc** [comment]
i reworded a commit title so the developer will not be confused by the `cast_batch` function name (was `cast_many` in the title).

File: test/gen_batch_server_SUITE.erl (hunk 1 above)
Comment (benoitc) [reply test/gen_batch_server_SUITE.erl]: i can remove both messages if @kjnilsson is OK . Just let me know :)

Comment (benoitc) [reply test/gen_batch_server_SUITE.erl]: @michaelklishin i've removed both messages in separate commit. if  @kjnilsson think the message needs to be kept i can revert 3f35b955b824c51a2b9df134dbb2abf2df1877b8 .

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
Thank you!

**Comment by kjnilsson** [comment]
Thanks this is great - I know there are a couple of places in Ra where batching write requests to the WAL can be done. I've pushed 0.7.0 to hex

**Pull request merged**